### PR TITLE
test: cover lifecycle safety-net gaps (#33)

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Adds deterministic tests for core-lifecycle paths that #33 flagged as having no coverage. Test-only; no production code changed.

## Coverage added

- **Status** — the genuinely-interrupted branch (process gone, no sentinel, no out file -> `failed`), which `TestStatusInterruptedAfterReboot` did not cover (it asserts the recovered-output `done` branch); and `errorSummary`'s UTF-8 boundary truncation when the trailing stderr exceeds `errTailBytes` and the cut lands mid-rune.
- **Gate** — `newGate` clamps a non-positive cap to 1; `release` underflow is absorbed so `inFlight` never goes negative (which would silently raise the cap).
- **buildAgyArgs** — pins the agy command line: `--print-timeout`, `--model`, repeated `--add-dir`, `--conversation`, and `-p` ordering, with optional flags omitted when empty.
- **Cancel** — the dead/stale-supervisor no-op and the `Load`-error path (only the live-supervisor path was covered).
- **config.Resolve** — the `AGY_MCP_STATE_DIR` override, `AGY_MCP_DEFAULT_MODEL`, the `JobTTL` default (a regression to 0 would silently disable GC), and the agy-not-on-PATH error.
- **serve_http** — assert the exact registered tool set (names + count) instead of a `>= 5` lower bound that would miss a dropped or renamed tool.

## Deferred (need a production change, not pure test additions)

Left as a follow-up so this PR stays test-only:
- Supervisor SIGKILL-escalation-after-killGrace test (needs `killGrace` injectable) and the spawn-failure (127) path.
- GC's keep-an-expired-but-alive-job invariant (needs a live process in the test).
- Platform-gating the `mcptools` tests so `go test ./...` passes on macOS/Windows (CI gates tests to ubuntu today).

## Test plan

- [x] `go test -race ./...` green; `go vet` clean; `golangci-lint` 0 issues; `gofmt` clean

Refs #33

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage: config resolution (state-dir override, default model and job TTL, behavior when agy is absent), job cancellation edge cases and load-error handling, concurrency gate edge cases (non‑positive caps and release underflow), argument construction for job starts, status reporting edge cases (interrupted/no-output and UTF‑8-safe truncation), and exact tool-listing verification for the HTTP tools endpoint.

### CODE REVIEW

File: internal/config/config_test.go [hunk 1]
```diff
@@ -3,10 +3,69 @@ package config
...
 
+// fakeAgyOnPath puts an executable named "agy" on PATH and clears the path
+// override, so Resolve's PATH lookup succeeds without depending on a real agy.
+func fakeAgyOnPath(t *testing.T) {
+	t.Helper()
+	t.Setenv("AGY_MCP_AGY_PATH", "")
+	dir := t.TempDir()
+	if err := os.WriteFile(filepath.Join(dir, "agy"), []byte("#!/bin/sh\n"), 0o755); err != nil {
+		t.Fatal(err)
+	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/config/config_test.go:19]: Good cross-platform catch, fixed in 99d2d00. The helper now also writes an empty agy.exe so exec.LookPath resolves it via PATHEXT on Windows; the extra file is ignored on Unix where the executable agy is found directly.
